### DISCUSSION TITLE
Dublicates QSO ion import?

### DISCUSSION THREAD

**Discussion by R1BLH** [magicbug/Cloudlog#1797] [General]
@magicbug 
It looks like there is a bug with dublicates QSO on the import ADIF file.
Look on tyhe screenshot 1. 

![2022-11-19_22-55-20](https://user-images.githubusercontent.com/16667173/202871194-c2514e78-0393-4a09-9890-5e7f4dc2cdee.png)
Here I deleted some of the dublicates (callisigns which started from digit). Now only 96 dublicates had been found.

After that i imported ADIF and got a message about dublicates.
Some of the lines of this message:
Date/Time: 2022-11-17 19:34:30 Callsign: 4X1UF Band: 40m Duplicate
Date/Time: 2022-11-16 23:08:15 Callsign: 4X6YA Band: 80m Duplicate
Date/Time: 2022-11-16 07:24:00 Callsign: 9A2KS Band: 15m Duplicate
Date/Time: 2022-11-15 14:40:45 Callsign: 9A2KS Band: 20m Duplicate
Date/Time: 2022-11-15 14:28:42 Callsign: 9A7R Band: 17m Duplicate
Date/Time: 2022-11-14 10:29:15 Callsign: 9A5CW Band: 12m Duplicate

After that i made search for dublicates, result are on the screenshot 2.
![2022-11-19_22-57-26](https://user-images.githubusercontent.com/16667173/202871214-bb47365b-32da-4497-b4a3-29f85df5922f.png)
All previously deleted lines are in the list again. Now 106 dublicates had been found again.

**Comment by magicbug** [comment]
If you delete duplicates the ADIF import has no idea that has happened so it will add them back.. not sure if @AndreasK79 considered this

**Reply by AndreasK79** [reply]
I don't get what you mean @magicbug. Duplicates are duplicates. If you import them back, they will be back.

**Reply by R1BLH** [reply]
> Duplicates are duplicates. If you import them back, they will be back.
@AndreasK79 
If the "Skip Duplicate QSO Check" option was unchecked when importing ADIF, does that mean only information about duplicates, not skipping them?
Is it possible to add the option to skip duplicates when importing ADIF?

**Reply by R1BLH** [reply]
@magicbug another question about duplicates.
I did a search and removed all the duplicates that were in the search results, then while loading the QSL pictures I found some duplicates manually, these duplicates were not shown in the duplicate search results.
I checked the list of duplicates in the message from the ADIF import and found that the duplicate QSO was not there.
Have you an idea why dublicate search did not found them all?

**Reply by AndreasK79** [reply]
The skip duplicates on import, checks exact time/date/band/mode/call. The duplicate search checks for duplicates within 30 minutes time difference. 

They are built a bit different.

**Reply by R1BLH** [reply]
> The skip duplicates on import, checks exact time/date/band/mode/call. The duplicate search checks for duplicates within 30 minutes time difference.

Please look on the screenshots with dublicate examples:  time/date/band/mode/call are the same.

![2022-11-20_03-16-49](https://user-images.githubusercontent.com/16667173/202915012-ad2e31c7-bf5e-445b-b36d-ffd7b97bf964.png)

![2022-11-20_03-38-42](https://user-images.githubusercontent.com/16667173/202915110-36ae75be-ecf4-41ed-bf13-c17734f9670a.png)
![2022-11-20_03-26-06](https://user-images.githubusercontent.com/16667173/202915120-a6faa67f-4a32-407a-8463-040ac0bdc33d.png)

**Reply by AndreasK79** [reply]
@cats-shadow Thanks for the examples. These were found by the search duplicate check, but there were not stopped by the duplicate check on import?

**Reply by AndreasK79** [reply]
@cats-shadow do you have the ADIF-file for these QSOs? Can you check the time and see if the seconds are different? The import checks on exact time.

**Reply by R1BLH** [reply]
> @cats-shadow Thanks for the examples. These were found by the search duplicate check, but there were not stopped by the duplicate check on import?

These examples were found by duplicate check (the were in the list of dublicates after import) on import and not were found by search for duplicates in web interface, i found them manually during QSL images upload.

**Reply by R1BLH** [reply]
> @cats-shadow do you have the ADIF-file for these QSOs? Can you check the time and see if the seconds are different? The import checks on exact time.

Yes time is exactly the same in the ADIF-file and in the CLoudlog. But i already wiped dublicates from Cloudlog.
I could recheck it, if i would find new cases, i could give more detailed information.

**Reply by R1BLH** [reply]
> The skip duplicates on import, checks exact time/date/band/mode/call. 

@AndreasK79 
Could you clarify this point, please? Are the QSOs from the list of duplicates, shown after the import of ADIF, imported into the cloudlog or not?

**Reply by AndreasK79** [reply]
If you get the duplicate message on the QSO, it should not be imported into the database. At least that is what the code is supposed to do when skip duplicates is not checked.
